brandingbrand/react-native-web
# PCR-916 - server side rendering

**Ticket:** https://jira.brandingbrand.com/browse/PCR-916
@brandingbrand/product 

## Description
This should probably be in its own branch and not in lastGood, but I could not create a new branch in this repo, so you can make a new branch based on lastGood then modify base fork of this PR

* merge master from https://github.com/necolas/react-native-web/ into lastGood to get latest changes, mostly SSR fix https://github.com/brandingbrand/react-native-web/commit/240cf7e05f05a622621d0ee5f58380a35d28ab44
* cherry pick back in a commit that got reverted so Images are included server side https://github.com/necolas/react-native-web/issues/543
* add additional check so it doesn't use window.Image server side

## Test
* use with my party city test branch https://github.com/NickBurkhartBB/partycity.rn/tree/PCR-916
* setup project running all of these `npm i` `npm run tsc` `cd web-server` `npm i` `npm run build-client` `npm run build-server` `npm run build-client` `npm run start-server`
* go to product page(it's the only one I setup redux store modifications) http://localhost:3000/shenanigans-st.-patricks-day-t-shirt-P765906.html?cgid=st-patricks-day-shirts&start=1
*ER: see page html
* regression: run with party city as is
* ER: doesn't explode
